src/components/modal/ModalStepper.tsx

```diff
@@ -1,19 +1,18 @@
 import * as React from "react";
 import Box from "@mui/material/Box";
-import Stepper from "@mui/material/Stepper";
-import Step from "@mui/material/Step";
-import StepLabel from "@mui/material/StepLabel";
-import { useModalContext } from "./context";
+import { Stepper, Step, StepLabel } from "@mui/material";
+import { ContextProps, useModalContext } from "./context";
 
-type Props = {
+interface Props {
   labels: string[];
-};
+}
 
 export const ModalStepper = ({ labels }: Props) => {
-  const stepIndex = useModalContext();
+  const { stepIndex } = useModalContext() as ContextProps;
```

Comment:
Anche in questo caso stai castando e forzando il type. 

Le due opzioni opzioni corrette sono 

```
const {stepIndex} = useModalContext() || { }
```

```
const ctx = useModalContext()
// Poi trovi la pops sotto
ctx?.stepIndex
```

In entrambi i casi nel componente dovrai andare a controllare che il valore di `stepIndex` esista e solo in quel caso fare il render del component o ritornare `null`.